docs/source/tutorials/qmd/07-error-correction.qmd

```diff
@@ -0,0 +1,623 @@
+mu_x = lsl.Var.new_param(0.0, distribution = mu_x_prior, name = "mu_x")
+
+# Define prior distribution for x
+x_prior_dist = lsl.Dist(tfd.Normal, loc = mu_x, scale = tau2_x)
+
+# Estimate x using the mean of replicates and assign a prior distribution
+x_estimated = lsl.Var.new_param(jnp.expand_dims(x_tilde.mean(axis = 1), -1), # initial estimation is the mean of the replicates
+                                distribution = x_prior_dist,
+                                name = "x_estimated")
+
+# Define likelihood model for measurement error
+measurement_dist = lsl.Dist(
+    tfd.MultivariateNormalFullCovariance,
+    loc = x_estimated,
+    covariance_matrix = sigma
+    )
+
+# Define x_tilde (observed replicates) as observed data in the probabilistic model
+x_tilde_var = lsl.Var.new_obs(
+    value = x_tilde,
+    distribution = measurement_dist,
+    name = "x_tilde"
+)
+```
+## The Distributional Regression Model
+
+Following that we will set up our distributional regression model using P-splines. Assuming a Gaussian with $N(\mu_i, \sigma_i^{2})$ for the response, the regression parameters are linked to the model parameters by $\mu_i = \eta^{\mu}(x_i)$ and $\sigma_i^{2} = exp(\eta^{\sigma^{2}}(x_i))$, where $exp(\cdot)$ ensures positivity of the scale parameter. The predictors are then given by:
+
+$$
+\eta^{\mu}(x_i) = \beta_{0}^{\mu} + f^{\mu}(x_i)
+$$
+
+and
+
+$$
+\eta^{\sigma^{2}}(x_i) = \beta_{0}^{\sigma^{2}} + f^{\sigma^{2}}(x_i),
+$$
+
+$\beta_{0}$ being an intercept term, and
+
+$$
+f(\cdot) = \mathbf{b}_i'\boldsymbol{\beta}.
+$$
+
+$\mathbf{b}_i'$ is defined as the vector of the evaluations of the basis function and $\boldsymbol{\beta}$ represents the corresponding vector of the basis coefficients. In our implementation we set $\beta_0 = 0$. We generate equidistant knots based on the observed $x$-values and construct cubic B-spline basis matrices for both functions using the Liesel {func}`.basis_matrix()` implementation. The number of knots is set to 20.
+
+```{python}
+#| label: P-splines
+# Define Number of knots/spline parameters to use
+n_param_splines = 20
+
+# Generate equidistant knots for spline basis functions based on the mean of replicates (x_tilde)
+knots = equidistant_knots(jnp.expand_dims(x_tilde.mean(axis = 1), -1), n_param = n_param_splines, order = 3)
+
+# Compute spline basis matrix for modeling the mean function
+basis_matrix_var_mu = lsl.Var.new_calc(
+    lambda x: splines.basis_matrix(
+        x.squeeze(),
+        knots = knots,    # Use precomputed knots
+        order = 3,        # Cubic spline basis
+        outer_ok = True),
+    x = x_estimated,
+    name = "basis_matrix_mu"
+)
+
+# Compute spline basis matrix for modeling the scale
+basis_matrix_var_scale = lsl.Var.new_calc(
+    lambda x: splines.basis_matrix(
+        x.squeeze(),
+        knots = knots,
+        order = 3,
+        outer_ok = True),
+    x = x_estimated,
+    name = "basis_matrix_scale"
+)
```

Comment:
The basis matrix can be the same for both the location and scale, so no need to use two variables here. That should actually reduce computational burden.